relay-kafka/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+pub mod config;
+#[cfg(feature = "processing")]
+mod producer;
+#[cfg(feature = "processing")]
+mod utils;
```

Comment:
Since this module just contains producer utilities, why don't we merge them into the `producer` module? That would also be quite nice if we rename the feature to `producer`.